File: 2711-difference-of-number-of-distinct-values-on-diagonals/2711-difference-of-number-of-distinct-values-on-diagonals.py
class Solution:
    def differenceOfDistinctValues(self, grid):
        ans = [[0] * len(grid[0]) for _ in range(len(grid))]
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                a = set()
                b = set()
                for k in range(1, min(i + 1, j + 1)):
                    a.add(grid[i - k][j - k])
                for k in range(1, min(len(grid) - i, len(grid[0]) - j)):
                    b.add(grid[i + k][j + k])
                ans[i][j] = abs(len(a) - len(b))
        return ans

File: 2711-difference-of-number-of-distinct-values-on-diagonals/2711-difference-of-number-of-distinct-values-on-diagonals.cpp
class Solution {
public:
    vector<vector<int>> differenceOfDistinctValues(vector<vector<int>>& grid) {
        vector<vector<int> > ans(grid.size(),vector<int>(grid[0].size()));
        for(int i = 0;i<grid.size();i++){
            for(int j = 0;j<grid[0].size();j++){
                set<int> a,b;
                for(int k =1;i-k>=0&&j-k>=0;k++)a.insert(grid[i-k][j-k]);
                for(int k =1;i+k<grid.size()&&j+k<grid[0].size();k++)b.insert(grid[i+k][j+k]);
                ans[i][j]=abs((int)(a.size()-b.size()));    
            }
        }
        return ans;
    }
};